framework/web/AssetManager.php

```diff
@@ -576,6 +587,10 @@ public function getPublishedUrl($path)
      */
     protected function hash($path)
     {
+        if (!empty($this->hashCallback)) {
+            return call_user_func($this->hashCallback, $path);
+        }
+        $path = (pathinfo($path, PATHINFO_EXTENSION) === '' ? $path : dirname($path)) . filemtime($path);
```

Comment:
I suppose current implementation is fine: particular hash callback may not need such logic. Still perhaps note about `$path` can lead to directory or file should be added to the doc commets at `hashCallback` declaration.